Content.Shared/Humanoid/Markings/MarkingColoring.cs

```diff
@@ -0,0 +1,227 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Humanoid.Markings;
+
+[Serializable, NetSerializable]
+public enum MarkingColoringType : byte
+{
+    SimpleColor,
+    SkinColor,
+    HairColor,
+    FacialHairColor,
+    AnyHairColor,
+    Tattoo,
+    EyeColor
+}
+
+
+[DataDefinition]
+[Serializable, NetSerializable]
+public sealed class MarkingColors
+{
+    /// <summary>
+    /// Coloring properties that will be used on any unspecified layer
+    /// </summary>
+    [DataField("default", true)]
+    public ColoringProperties Default { get; } = new();
+
+    /// <summary>
+    /// Layers with their own coloring type and properties
+    /// </summary>
+    [DataField("layers", true)]
+    public Dictionary<string, ColoringProperties>? Layers;
+}
+
+/// <summary>
+/// Properties for coloring. 
+/// </summary>
+[DataDefinition]
+[Serializable, NetSerializable]
+public sealed class ColoringProperties
+{
+    /// <summary>
+    /// Coloring type that will be used on that layer
+    /// </summary>
+    [DataField("type", true, required: true)]
+    public MarkingColoringType Type { get; } = MarkingColoringType.SkinColor;
+
+    /// <summary>
+    /// Color that will be used if coloring type can't be performed or used "SimpleColor" type
+    /// </summary>
+    [DataField("color", true)]
+    public Color Color { get; } = Color.White;
+
+    /// <summary>
+    /// Makes output color negative if true (in e.x. Different Eyes)
+    /// </summary>
+    [DataField("negative", true)]
+    public bool Negative { get; } = false;
+}
+
+
+public static class MarkingColoring
+{
+    /// <summary>
+    /// Returns list of colors for marking layers
+    /// </summary>
+    public static List<Color> GetMarkingLayerColors
+    (
+        MarkingPrototype prototype,
+        Color? skinColor,
+        Color? eyeColor,
+        Color? hairColor,
+        Color? facialHairColor
+    )
+    {
+        List<Color> colors = new List<Color>();
+
+        // Coloring from default properties
+        Color default_color = MarkingColoring.GetMarkingColor(
```

Comment:
incorrect variable naming, should be `camelCase` (e.g., `defaultColor` instead of `default_color`)